### ISSUE TITLE
Repository.Remote fails if the repository has branches that checked out from pull-request

### ISSUE THREAD

**Issue by kyoh86** [go-git/go-git#331]
## Summary

Using GitHub CLI `gh` ( https://github.com/cli/cli ), I often checkout pull-request.

```console
gh pr checkout 330
```

But go-git fails to parse the config in the repository that has a branch created by `gh`.

## To reproduce:

1. Prepare a repository

```console
$ git clone https://github.com/go-git/go-git
$ gh pr checkout 330
```

2. Try to get the remote URLs from the repository with `go-git`

```go
	repo, err := git.PlainOpen("./go-git")
	if err != nil {
		return nil, fmt.Errorf("open local repository: %w", err)
	}
	remote, err := repo.Remote("origin")
	if err != nil {
		return nil, fmt.Errorf("get remote %s: %w", name, err)
	}
	return remote.Config().URLs, nil
```

It fails.

## Supplement

`gh pr checkout xxx` creates a branch like below.

```ini
[branch "foo/bar"]
	remote = origin
	merge = refs/pull/330/head
```

`go-git` says "branch config: invalid merge".

https://github.com/go-git/go-git/blob/da810275bf682d29a530ed819aff175f47bd7634/config/branch.go#L39-L41
https://github.com/go-git/go-git/blob/da810275bf682d29a530ed819aff175f47bd7634/config/branch.go#L12
https://github.com/go-git/go-git/blob/da810275bf682d29a530ed819aff175f47bd7634/plumbing/reference.go#L89-L91
https://github.com/go-git/go-git/blob/da810275bf682d29a530ed819aff175f47bd7634/plumbing/reference.go#L11

**Comment by ordovicia** [comment]
Hi, @mcuadros.
I would like to work on a fix for this issue.
What do you think is the best solution?
Removing or relaxing the `IsBranch()` validation would be a quick fix, but it allows go-git to accept invalid `.git/config`.
On the other hand, adding special treatment for `refs/pull/*` format would reduce go-git maintenability.

**Comment by kyoh86** [comment]
@mcuadros how about this?

**Comment by w568w** [comment]
Bump it. It is really an usual scenario that a local repository contains branches from `refs/pull/*`. I am not a Go developer but just checking a branch merge against `"refs/heads/"` prefix to verify its availability does not seem correct. 

Some downstream projects had to [switch to their own fork](https://github.com/jesseduffield/lazygit/pull/1037) to fix this problem.

**Comment by pjbgf** [comment]
I used the steps to mentioned above and could not reproduce the problem:

```
$ git clone https://github.com/go-git/go-git
$ cd /tmp/gogit
$ gh pr checkout 1045
$ go run _examples/open/main.go /tmp/gogit
git rev-list HEAD --count
2153
https://github.com/go-git/go-git
Current Branch:%!(EXTRA string=refs/heads/fix-amend-with-changes)
```

Please note that the `open` example was modified to include:
```golang
	remote, err := r.Remote("origin")
	CheckIfError(err)

	Info(strings.Join(remote.Config().URLs, " "))

	head, err := r.Head()
	CheckIfError(err)

	Info("Current Branch:", head.Name().String())
```

Anyone still experiencing this with the version in `master`?

**Comment by kyoh86** [comment]
@pjgbf I still experiencing it.
What does `.git/config` look like after running `gh pr checkout 330`?
I have this: `.git/config`:

```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
[remote "origin"]
	url = https://github.com/go-git/go-git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[branch "fix-323-fix-broken-large-ofs-delta"]
	remote = origin
	pushRemote = origin
	merge = refs/pull/330/head
```

**Comment by pjbgf** [comment]
@kyoh86 my `go-git` repository has a `.git/config` with dozens of entries created by `gh pr checkout`, but I still could not reproduce this. Anything on your `~/.gitconfig` that could be influencing here?

Otherwise, any additional information that could help us reliably reproduce this would be great.

**Comment by kyoh86** [comment]
@pjbgf 
Why can't you provide your `.git/config` as I have provided it...?
My ~/.gitconfig and ~/.config/git/config are empty.

**Comment by w568w** [comment]
I can also reproduce the bug by strictly following @pjbgf's steps. So you are not alone ;)

```
$ go run _examples/open/main.go /tmp/go-git
git rev-list HEAD --count
2153
error: branch config: invalid merge
exit status 1
```

My `.git/config`:

```
[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
[remote "origin"]
        url = https://github.com/go-git/go-git
        fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
        remote = origin
        merge = refs/heads/master
[branch "fix-amend-with-changes"]
        remote = origin
        pushRemote = origin
        merge = refs/pull/1045/head
```

**Comment by pjbgf** [comment]
@w568w I missed adding `/tmp/go-git` on the clone operation of the steps to reproduce, updated that now for future reference.

I now managed to reproduce the issue. What I noticed is that `gh pr checkout` behaves differently at times, on the .git/config I have that works, any new `gh pr checkout` is added using solely the target fork information:

```
[branch "rodrigocam/master"]
	remote = git@github.com:rodrigocam/go-git.git
	pushRemote = git@github.com:rodrigocam/go-git.git
	merge = refs/heads/master
```

The one that doesn't, aligns with what was already reported above. Here are [both configs](https://gist.github.com/pjbgf/501a66a9d3865af024ad402c05d98958) I have.

But overall, I am more than happy to review a PR that fixes this.
